collect_app/src/main/java/org/odk/collect/android/map/MbtilesFile.java

```diff
@@ -0,0 +1,146 @@
+package org.odk.collect.android.map;
+
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteException;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import androidx.annotation.NonNull;
+import timber.log.Timber;
+
+/**
+ * This class provides access to the metadata and tiles in a .mbtiles file.
+ * An .mbtiles file is a SQLite database file containing specific tables and
+ * columns, including tiles that may contain raster images or vector geometry.
+ * See https://github.com/mapbox/mbtiles-spec for the detailed specification.
+ */
+public class MbtilesFile implements Closeable, TileHttpServer.TileSource {
+    public enum Type { RASTER, VECTOR }
+
+    protected File file;
+    protected SQLiteDatabase db;
+    protected String format;
+    protected Type type;
+    protected String contentType = "application/octet-stream";
+    protected String contentEncoding = "identity";
+
+    public MbtilesFile(File file) throws SQLiteException, UnsupportedFormatException {
+        this.file = file;
+        db = SQLiteDatabase.openOrCreateDatabase(file, null);
```

Comment:
Probably the future goal would be to use this class for the Google Maps offline tiles (or offiline tiles as I believe they are called there) as well, right? How do we keep track of these aspirations? Or do you think it's obvious enough that when someone gives `MapHelper` some love, s/he will see that there's a nice abstraction to use here?